test/object-store/sync/app.cpp

```diff
@@ -3053,6 +3062,40 @@ TEST_CASE("app: sync integration", "[sync][app]") {
         }
     }
 
+    SECTION("pausing a session does not hold the DB open") {
+        SyncTestFile config(app, partition, schema);
+        DBRef dbref;
+        std::shared_ptr<SyncSession> sync_sess_ext_ref;
+        {
+            auto realm = Realm::get_shared_realm(config);
+            wait_for_download(*realm);
+
+            auto state = realm->sync_session()->state();
+            REQUIRE(state == SyncSession::State::Active);
+
+            sync_sess_ext_ref = realm->sync_session()->external_reference();
+            dbref = TestHelper::get_db(*realm);
+            REQUIRE(dbref.use_count() == 4);
```

Comment:
I'm assuming this refcount of 4 is for: the local dbref, the 2 sync_session references and the sync::Session reference.

Reply:
1 for the local dbref. 1 for the realm coordinator. 1 for the SyncSession and one for the SessionWrapper.

Reply:
Ah, I forgot about the realm coordinator